backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/objcTypes.kt

```diff
@@ -51,16 +52,45 @@ class ObjCClassType(
     }
 }
 
-class ObjCGenericTypeDeclaration(
-        val typeParameterDescriptor: TypeParameterDescriptor,
-        val namer: ObjCExportNamer
-) : ObjCNonNullReferenceType() {
+sealed class ObjCGenericType : ObjCNonNullReferenceType() {
+    abstract val variance: Variance
```

Comment:
`Variance` is Kotlin variance. Obj-C type can't have Kotlin variance.

Reply:
Introduced typed `ObjCVariance` enum.